<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ScatterPlotPropertiesWidget</class>
 <widget class="QWidget" name="ScatterPlotPropertiesWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>300</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Scatter Plot Properties</string>
  </property>
  <layout class="QVBoxLayout" name="main_layout">
   <property name="spacing">
    <number>10</number>
   </property>
   <property name="leftMargin">
    <number>5</number>
   </property>
   <property name="topMargin">
    <number>5</number>
   </property>
   <property name="rightMargin">
    <number>5</number>
   </property>
   <property name="bottomMargin">
    <number>5</number>
   </property>
   <item>
    <widget class="QGroupBox" name="x_axis_group">
     <property name="title">
      <string>X-Axis Data</string>
     </property>
     <layout class="QVBoxLayout" name="x_axis_layout">
      <item>
       <layout class="QHBoxLayout" name="x_axis_selection_layout">
        <item>
         <widget class="QLabel" name="x_axis_label">
          <property name="text">
           <string>Data Source:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="x_axis_combo">
          <property name="minimumWidth">
           <number>150</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="x_axis_info_label">
        <property name="text">
         <string>Select a data source for the X-axis</string>
        </property>
        <property name="styleSheet">
         <string notr="true">QLabel { 
color: #555; 
font-size: 11px; 
padding: 8px; 
background-color: #f0f0f0; 
border: 1px solid #ccc; 
border-radius: 4px; 
}</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="y_axis_group">
     <property name="title">
      <string>Y-Axis Data</string>
     </property>
     <layout class="QVBoxLayout" name="y_axis_layout">
      <item>
       <layout class="QHBoxLayout" name="y_axis_selection_layout">
        <item>
         <widget class="QLabel" name="y_axis_label">
          <property name="text">
           <string>Data Source:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="y_axis_combo">
          <property name="minimumWidth">
           <number>150</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="y_axis_info_label">
        <property name="text">
         <string>Select a data source for the Y-axis</string>
        </property>
        <property name="styleSheet">
         <string notr="true">QLabel { 
color: #555; 
font-size: 11px; 
padding: 8px; 
background-color: #f0f0f0; 
border: 1px solid #ccc; 
border-radius: 4px; 
}</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="filter_group">
     <property name="title">
      <string>Filter By Range</string>
     </property>
     <layout class="QVBoxLayout" name="filter_layout">
      <item>
       <widget class="QCheckBox" name="filter_enabled_checkbox">
        <property name="text"><string>Enable Filter</string></property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="filter_sel_layout">
        <item><widget class="QLabel" name="filter_col_label"><property name="text"><string>Table.Column:</string></property></widget></item>
        <item><widget class="QComboBox" name="filter_column_combo"><property name="minimumWidth"><number>200</number></property></widget></item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="filter_cond_layout">
        <item><widget class="QComboBox" name="filter_comparator_combo">
          <item><property name="text"><string>&lt;</string></property></item>
          <item><property name="text"><string>&lt;=</string></property></item>
          <item><property name="text"><string>&gt;</string></property></item>
          <item><property name="text"><string>&gt;=</string></property></item>
          <item><property name="text"><string>==</string></property></item>
          <item><property name="text"><string>!=</string></property></item>
        </widget></item>
        <item><widget class="QDoubleSpinBox" name="filter_value_spin"><property name="minimum"><double>-1e12</double></property><property name="maximum"><double>1e12</double></property><property name="decimals"><number>6</number></property></widget></item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="color_group">
     <property name="title">
      <string>Color By Feature</string>
     </property>
     <layout class="QVBoxLayout" name="color_layout">
      <item><widget class="QCheckBox" name="color_enabled_checkbox"><property name="text"><string>Enable Color Mapping</string></property></widget></item>
      <item>
       <layout class="QHBoxLayout" name="color_sel_layout">
        <item><widget class="QLabel" name="color_col_label"><property name="text"><string>Table.Column:</string></property></widget></item>
        <item><widget class="QComboBox" name="color_column_combo"><property name="minimumWidth"><number>200</number></property></widget></item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="color_mode_layout">
        <item><widget class="QLabel" name="color_mode_label"><property name="text"><string>Mode:</string></property></widget></item>
        <item><widget class="QComboBox" name="color_mode_combo">
          <item><property name="text"><string>Bool</string></property></item>
          <item><property name="text"><string>Int</string></property></item>
          <item><property name="text"><string>Continuous</string></property></item>
        </widget></item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="color_discrete_layout">
        <item><widget class="QLabel" name="color_false_label"><property name="text"><string>False/0 idx:</string></property></widget></item>
        <item><widget class="QSpinBox" name="color_false_spin"><property name="minimum"><number>0</number></property><property name="maximum"><number>31</number></property></widget></item>
        <item><widget class="QLabel" name="color_true_label"><property name="text"><string>True/1 idx:</string></property></widget></item>
        <item><widget class="QSpinBox" name="color_true_spin"><property name="minimum"><number>0</number></property><property name="maximum"><number>31</number></property></widget></item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="color_continuous_layout">
        <item><widget class="QLabel" name="color_min_label"><property name="text"><string>Min:</string></property></widget></item>
        <item><widget class="QDoubleSpinBox" name="color_min_spin"><property name="minimum"><double>-1e12</double></property><property name="maximum"><double>1e12</double></property></widget></item>
        <item><widget class="QLabel" name="color_max_label"><property name="text"><string>Max:</string></property></widget></item>
        <item><widget class="QDoubleSpinBox" name="color_max_spin"><property name="minimum"><double>-1e12</double></property><property name="maximum"><double>1e12</double></property></widget></item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="visualization_group">
     <property name="title">
      <string>Visualization Settings</string>
     </property>
     <layout class="QVBoxLayout" name="vis_layout">
      <item>
       <layout class="QHBoxLayout" name="point_size_layout">
        <item>
         <widget class="QLabel" name="point_size_label">
          <property name="text">
           <string>Point Size:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="point_size_spinbox">
          <property name="suffix">
           <string>px</string>
          </property>
          <property name="minimum">
           <double>0.500000000000000</double>
          </property>
          <property name="maximum">
           <double>20.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.500000000000000</double>
          </property>
          <property name="value">
           <double>3.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_point_size">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="point_color_layout">
        <item>
         <widget class="QLabel" name="point_color_label">
          <property name="text">
           <string>Point Color:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="point_color_button">
          <property name="minimumSize">
           <size>
            <width>60</width>
            <height>25</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>60</width>
            <height>25</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="styleSheet">
           <string notr="true">QPushButton { 
background-color: #3268a8; 
border: 1px solid #ccc; 
border-radius: 3px; 
}</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_point_color">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="show_grid_checkbox">
        <property name="text">
         <string>Show Grid</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="show_legend_checkbox">
        <property name="text">
         <string>Show Legend</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="view_help_label">
        <property name="text">
         <string>View Controls:
- Mouse Wheel: Zoom (both axes)
- Ctrl + Wheel: Zoom X only
- Shift + Wheel: Zoom Y only
- Left-Drag: Pan
- Alt + Left-Drag: Box-zoom to rectangle</string>
        </property>
        <property name="styleSheet">
         <string notr="true">QLabel { 
color: #555; 
font-size: 11px; 
padding: 8px; 
background-color: #f0f0f0; 
border: 1px solid #ccc; 
border-radius: 4px; 
}</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="coordinate_range_group">
     <property name="title">
      <string>Coordinate Range</string>
     </property>
     <layout class="QVBoxLayout" name="coordinate_range_layout">
      <item>
       <layout class="QHBoxLayout" name="x_range_layout">
        <item>
         <widget class="QLabel" name="x_range_label">
          <property name="text">
           <string>X Range:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="x_range_value">
          <property name="text">
           <string>No data</string>
          </property>
          <property name="styleSheet">
           <string notr="true">QLabel { 
color: #555; 
font-size: 11px; 
padding: 4px; 
background-color: #f8f8f8; 
border: 1px solid #ddd; 
border-radius: 3px; 
}</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="y_range_layout">
        <item>
         <widget class="QLabel" name="y_range_label">
          <property name="text">
           <string>Y Range:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="y_range_value">
          <property name="text">
           <string>No data</string>
          </property>
          <property name="styleSheet">
           <string notr="true">QLabel { 
color: #555; 
font-size: 11px; 
padding: 4px; 
background-color: #f8f8f8; 
border: 1px solid #ddd; 
border-radius: 3px; 
}</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="coordinate_info_label">
        <property name="text">
         <string>Coordinate ranges update when you zoom or pan the plot</string>
        </property>
        <property name="styleSheet">
         <string notr="true">QLabel { 
color: #666; 
font-size: 10px; 
font-style: italic; 
padding: 4px; 
}</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui> 